packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -992,13 +1001,25 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
   ///  * [RenderObjectToWidgetAdapter.attachToRenderTree], which inflates a
   ///    widget and attaches it to the render tree.
   void attachRootWidget(Widget rootWidget) {
-    final bool isBootstrapFrame = rootElement == null;
-    _readyToProduceFrames = true;
-    _rootElement = RenderObjectToWidgetAdapter<RenderBox>(
-      container: renderView,
+    attachToBuildOwner(RootWidget(
       debugShortDescription: '[root]',
       child: rootWidget,
-    ).attachToRenderTree(buildOwner!, rootElement as RenderObjectToWidgetElement<RenderBox>?);
+    ));
+  }
+
+  /// Called by [attachRootWidget] to attached the provided [RootWidget] to
+  /// the [buildOwner].
+  ///
+  /// This creates the [rootElement], if necessary, or re-uses an existing one.
+  ///
+  /// This method is rarely called directly, but it can be useful in tests to
```

Comment:
It is used by the testing framework (flutter_test), which is technically outside of a test test, which is why `@visibleForTesting` doesn't work here...

Reply:
Okay, so it doesn't help with actually using it where you want to use it, but it would provide a hint to people that it isn't expected to be used in production code.  You can always exempt it in the flutter_test code to run it.

Reply:
It is perfectly fine to use in production code, though, and some frameworks build on top of flutter may actually do so. There is nothing wrong with that. With this comment I only tried to discourage somebody, who casually stumbles over this to think that its a super-useful thing for them...

Reply:
Okay, no worries, thn.